transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -113,6 +114,7 @@ def __init__(self, port, on_scan=None, on_disconnect=None, passive=None, **kwarg
         self.count_lock = threading.Lock()
         self.connecting_count = 0
         self.maximum_connections = 0
+        self._conn_map = []
```

Comment:
There's a better data structure choice here than a list.

You want an `OrderedDict`.  It comes with `popitem(last=False)` to remove the oldest entry.  When setting a new entry, you can move it to end using `move_to_end` if it already exists:

```python
_conn_map = OrderedDict()

def _update_conn_map(conn_string, device_uuid):
    while len(_conn_map) >= ConnMapMaxSize:
        _conn_map.popitem(last=False)

    if conn_string in _conn_map:
        _conn_map.move_to_end(conn_string)
    else:
        _conn_map[conn_string] = device_uuid
```

Reply:
Gotcha... yeah I was looking at `OrderedDict` but i didn't see a `move_to_end` function. I must have been looking at an older version.

Reply:
The other standard option is just to `pop` and `add` so it ends up at the end anyway.